largest-series-product/example.rb

```diff
@@ -3,34 +3,46 @@
 class Series
   attr_reader :digits
   def initialize(numeric_string)
-    @digits = convert_to_digits(numeric_string)
+    @digits = numeric_string
   end
 
   def largest_product(length)
-    fail ArgumentError.new('Not enough digits') if length > digits.length
-    products = []
-    slices(length).each do |slice|
-      products << slice.inject(1) do |product, n|
-        product * n
-      end
-    end
-    products.sort.last
+    @length = length
+    validate_length
+    analyzer
   end
 
-  def slices(length)
-    result = []
-    i = -1
-    begin
-      i += 1
-      i2 = i + length - 1
-      result << digits[i..i2]
-    end while i2 < digits.size - 1
-    result
+  private
+
+  def validate_length
+    @length > digits.length and
+      fail(ArgumentError.new 'Not enough digits')
   end
 
-  private
+  def analyzer
```

Comment:
little naming nitpick: i think it's clearer when a method which returns something is named after the thing it returns, so maybe 'maximum' or if you want to show that the analysis is the important thing here I would say something like analysis_result. When you have a method called analyzer I would expect it to return a lambda (or object responding to call)